lib/matplotlib/font_manager.py

```diff
@@ -1203,10 +1194,19 @@ def score_weight(self, weight1, weight2):
         """
         Returns a match score between *weight1* and *weight2*.
 
-        The result is the absolute value of the difference between the
+        The result is 0.0 if both weight1 and weight 2 are given as strings
+        and have the same value.
+
+        Otherwise, the result is the absolute value of the difference between the
         CSS numeric values of *weight1* and *weight2*, normalized
-        between 0.0 and 1.0.
+        between 0.05 and 1.0.
         """
+
+        # exact match of the weight names (e.g. weight1 == weight2 == "regular")
+        if (isinstance(weight1, six.string_types) and
+            isinstance(weight2, six.string_types) and
```

Comment:
Please indent another 4 spaces; this should not be at the same level as the interior block.